src/odemis/acq/stream/_live.py

```diff
@@ -430,12 +430,12 @@ def _applyROI(self):
         """
         res, shift = self._computeROISettings(self.roi.value)
 
-        if (self._shiftebeam == MTD_EBEAM_SHIFT) and (self._beamshift is not None):
-            shift = tuple(s + c for s, c in zip(shift, self._beamshift))
-
-        # always in this order
+        # always in this order: resolution, then shift
         self._emitter.resolution.value = res
-        self._emitter.shift.value = shift
+        if self._shiftebeam == MTD_EBEAM_SHIFT:
+            if self._beamshift is not None:
+                shift = tuple(s + c for s, c in zip(shift, self._beamshift))
```

Comment:
this is always a typle of size 2 isn't it?
wouldn't :
shift = (shift[0] + self._beamshift[0], shift[1] + self._beamshift[1])
be more readable?

Reply:
I agree with you. It was a little bit of over-engineering. However, for this change, I didn't want to muddy the water. This line is still correct. So I'm leaving it as-is.
I'll push a second commit, with this clean-up.